src/Activation/useActivationNavigation.ts

```diff
@@ -0,0 +1,104 @@
+  | "ActivateExposureNotifications"
+  | "NotificationPermissions"
+  | "ActivationSummary"
+
+export const toScreen = (step: ActivationStep): ActivationStackScreen => {
+  switch (step) {
+    case "AcceptTermsOfService":
+      return ActivationStackScreens.AcceptTermsOfService
+    case "AnonymizedDataConsent":
+      return ActivationStackScreens.AnonymizedDataConsent
+    case "ActivateLocation":
+      return ActivationStackScreens.ActivateLocation
+    case "ActivateBluetooth":
+      return ActivationStackScreens.ActivateBluetooth
+    case "ActivateExposureNotifications":
+      return ActivationStackScreens.ActivateExposureNotifications
+    case "NotificationPermissions":
+      return ActivationStackScreens.NotificationPermissions
+    case "ActivationSummary":
+      return ActivationStackScreens.ActivationSummary
+  }
+}
+
+type ActivationNavigation = {
+  activationSteps: ActivationStep[]
+  goToNextScreenFrom: (currentStep: ActivationStep) => void
+}
+
+export const useActivationNavigation = (): ActivationNavigation => {
+  const {
+    displayAcceptTermsOfService,
+    enableProductAnalytics,
+  } = useConfigurationContext()
+  const navigation = useNavigation()
+  const { locationPermissions, isBluetoothOn } = usePermissionsContext()
+  const { completeOnboarding } = useOnboardingContext()
+
+  const environment = {
+    displayAcceptTermsOfService,
+    enableProductAnalytics,
+    locationPermissions,
+    isBluetoothOn,
+  }
+
+  const activationSteps = determineActivationSteps(environment)
+
+  const goToNextScreenFrom = (currentStep: ActivationStep) => {
+    const currentStepIndex: number | undefined = activationSteps.findIndex(
+      (step) => step === currentStep,
+    )
+    const nextStepIndex = currentStepIndex + 1
+    if (nextStepIndex < activationSteps.length) {
+      const nextStepName = activationSteps[nextStepIndex]
+      const nextScreen = toScreen(nextStepName)
+      navigation.navigate(nextScreen)
+    } else {
+      completeOnboarding()
+    }
+  }
+
+  return { activationSteps, goToNextScreenFrom }
+}
+
+export type Environment = {
+  locationPermissions: LocationPermissions
+  isBluetoothOn: boolean
+  displayAcceptTermsOfService: boolean
+  enableProductAnalytics: boolean
+}
+
+export const determineActivationSteps = ({
+  displayAcceptTermsOfService,
+  enableProductAnalytics,
+  locationPermissions,
+  isBluetoothOn,
+}: Environment): ActivationStep[] => {
+  const isLocationRequired = locationPermissions !== "NotRequired"
+
+  const activationSteps: ActivationStep[] = []
+
+  displayAcceptTermsOfService && activationSteps.push("AcceptTermsOfService")
+  enableProductAnalytics && activationSteps.push("AnonymizedDataConsent")
+  isLocationRequired && activationSteps.push("ActivateLocation")
+  !isBluetoothOn && activationSteps.push("ActivateBluetooth")
+  activationSteps.push("ActivateExposureNotifications")
+  Platform.OS === "ios" && activationSteps.push("NotificationPermissions")
```

Comment:
to keep this from needed to be mocked in a spec it might be nice to pas the os in as part of the env object.

Reply:
It doesn't appear to need to be mocked. Am I missing something?